get_ecg.py

```diff
@@ -0,0 +1,173 @@
+            try:
+                if dump[0]:
+                    max_peaks.pop(0)
+                del dump
+            except IndexError:
+                    # no peaks were found
+                    print("No peaks were found")
+
+        return self.total_peaks
+
+    def get_inst_hr(self):
+        """
+           Function that calculates inst heart rate every update_time seconds
+           by counting number of peaks within update_time
+
+           :return self.raw_bunches: divided the total_peaks into custom peaks
+           :rtype: nd array
+
+           """
+
+        num_of_peaks = []
+        if isinstance(self.update_time, float):
+            self.update_time = int(self.update_time)
+        for i in range(len(self.total_peaks)):
+            num_of_peaks.append(len(self.total_peaks[i]))
+        new_array = np.array(num_of_peaks)
+        inst_heart_rate = new_array / self.update_time
+        self.raw_bunches = inst_heart_rate * 60
+        return self.raw_bunches
+
+    def get_avghr(self):
+        """ returns avghr
+
+            Function gets avghr over user-specified minutes window.
+            Also gives clinical indication of tachy/bradycardia
+
+            :return avg_hr: Avg hr
+            :return status: Brady or tachycardia
+        """
+        mins = float(input('Please specify a time (in min) for averaging.'))
+        if isinstance(mins, float) is True or isinstance(mins, int) is True:
+            mins = mins
+        elif isinstance(mins, complex) is True:
+            raise ValueError('Please use real numbers.')
+
+        user_sec = mins * self.min2sec
+
+    # Must have number of groups be whole
+        if user_sec % self.update_time == 0:
+            group_num = int(user_sec/self.update_time)
+        # Taking an additional groups if user input is between groups
+        else:
+            group_num = int(math.floor(user_sec/self.update_time) + 1)
+
+        # Take Niranjana's grouping output and truncating to fit user input
+        if len(self.raw_bunches) <= group_num:  # if user time is > raw data
+            real_bunches = self.raw_bunches
+        else:
+            real_bunches = self.raw_bunches[0:group_num]
+
+    # Calculating the avghr
+        avg_hr = math.floor(sum(real_bunches)/len(real_bunches))
+
+    # Calculating brady-/tachycardia
+        if self.brady_threshold < avg_hr < self.tachy_threshold:
+            status = [2, 'You have a normal average heart rate '
+                         'over the period of {} minutes'.format(mins)]
+        elif avg_hr >= self.tachy_threshold:
+            status = [1, 'You have tachycardia over the '
+                         'period of {} minutes'.format(mins)]  # Tachy
+        elif avg_hr <= self.brady_threshold:
+            status = [0, 'You have bradycardia over '
+                         'the period of {} minutes'.format(mins)]  # Brady
+
+        return [status, avg_hr]
```

Comment:
What are `status` and `avg_hr` not attributes of this object?

Reply:
If these are 'final' variables that won't be used by any other code downstream, is it necessary to have them as attributes? Is it just better practice just in case other code is written later that will need to access them?